qiskit_ibm_runtime/session.py

```diff
@@ -124,7 +127,6 @@ def __init__(
         self._backend = backend
 
         self._setup_lock = Lock()
```

Comment:
This lock was added by https://github.com/Qiskit/qiskit-ibm-runtime/pull/1196 to make session thread safe, e.g. it enabled the following case

```
with Session(backend=backend), ThreadPoolExecutor() as executor:
    estimator1 = Estimator(options={"job_tags": ["cobyla"]})
    estimator2 = Estimator(options={"job_tags": ["nelder-mead"]})

    cobyla_result = executor.submit(minimize_thread, estimator1, "cobyla").result()
    nelder_mead_result = executor.submit(minimize_thread, estimator2, "nelder-mead").result()
```

I don't think it's needed anymore since session is now created prior to the threads start.